os/fs/driver/mtd/smart.c

```diff
@@ -4887,8 +4891,34 @@ static int smart_ioctl(FAR struct inode *inode, int cmd, unsigned long arg)
 
 	case BIOC_BULKERASE:
 		ret = MTD_IOCTL(dev->mtd, MTDIOC_BULKERASE, 0);
+
+		fdbg("Format Finished\n");
+		sleep(1);
 		goto ok_out;
-		
+	case BIOC_CORRUPTION :
+		sector = dev->sMap[SMART_FIRST_DIR_SECTOR];
+		header = (FAR struct smart_sect_header_s *)dev->rwbuffer;
+		ret = MTD_BREAD(dev->mtd, sector * dev->mtdBlksPerSector, dev->mtdBlksPerSector, (FAR uint8_t *)dev->rwbuffer);
+		if (ret != dev->mtdBlksPerSector) {
+			fdbg("Error reading root sector %d\n", sector);
+			ret = -EIO;
+			goto ok_out;
+		}
+#if CONFIG_SMARTFS_ERASEDSTATE == 0xFF
+		header->status = header->status & ~(SMART_STATUS_RELEASED | SMART_STATUS_COMMITTED);
+#else
+		header->status = header->status | SMART_STATUS_RELEASED | SMART_STATUS_COMMITTED;
+#endif
+		offset = sector * dev->mtdBlksPerSector * dev->geo.blocksize + offsetof(struct smart_sect_header_s, status);
+		ret = smart_bytewrite(dev, offset, 1, &header->status);
+		if (ret != 1) {
+			fdbg("Error %d releasing root sector %d\n" - ret, sector);
```

Comment:
Could you check this log?
`fdbg("Error %d releasing root sector %d\n" - ret, sector);`
I think
`fdbg("Error %d releasing root sector %d\n", -ret, sector);`